File: app/src/main/java/pl/kopsoft/pczplan/adapters/SemestersAdapter.java
package pl.kopsoft.pczplan.adapters;


import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import pl.kopsoft.pczplan.R;
import pl.kopsoft.pczplan.RecyclerViewClickListener;
import pl.kopsoft.pczplan.models.Semester;

public class SemestersAdapter extends RecyclerView.Adapter<SemestersAdapter.ViewHolder> {
    private List<Semester> semesters;
    private RecyclerViewClickListener clickListener;

    public SemestersAdapter(List<Semester> semesters, RecyclerViewClickListener listener) {
        this.semesters = semesters;
        clickListener = listener;
    }

    @NonNull
    @Override
    public SemestersAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View itemView = LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.semester_row, viewGroup, false);

        return new ViewHolder(itemView, clickListener);
    }

    @Override
    public void onBindViewHolder(@NonNull SemestersAdapter.ViewHolder viewHolder, int i) {
        viewHolder.TermName.setText(semesters.get(i).TermName);
    }

    @Override
    public int getItemCount() {
        return semesters.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        public TextView TermName;

        public ViewHolder(@NonNull View itemView, RecyclerViewClickListener listener) {
            super(itemView);
            clickListener = listener;
            TermName = itemView.findViewById(R.id.term_name);
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            clickListener.OnRecyclerItemClick(v, getAdapterPosition());
        }
    }
}

File: app/src/main/java/pl/kopsoft/pczplan/adapters/SemestersAdapter.kt
package pl.kopsoft.pczplan.adapters


import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.row_semester.view.*

import pl.kopsoft.pczplan.R
import pl.kopsoft.pczplan.interfaces.RecyclerViewClickListener
import pl.kopsoft.pczplan.models.Semester

class SemestersAdapter(
        private val semesters: List<Semester>,
        private var clickListener: RecyclerViewClickListener?) : RecyclerView.Adapter<SemestersAdapter.ViewHolder>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        val itemView = LayoutInflater.from(viewGroup.context).inflate(R.layout.row_semester, viewGroup, false)

        return ViewHolder(itemView, clickListener)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        viewHolder.termName.text = semesters[i].termName
    }

    override fun getItemCount(): Int {
        return semesters.size
    }

    inner class ViewHolder(itemView: View, private val listener: RecyclerViewClickListener?) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        val termName: TextView = itemView.term_name

        init {
            itemView.setOnClickListener(this)
        }

        override fun onClick(v: View) {
            listener?.onRecyclerItemClick(v, adapterPosition)
        }
    }
}
